### PR TITLE
remove string, use struct to build script request.  Add support for u…

### PR DESCRIPTION
**Pull request by zsiddique**
…psert and upsert with script

### CODE REVIEW

**Comment by zsiddique** [comment]
Looks like the test are broken for us.  I will look in to it next week but they seem no different then other branches

File: lib/corebulk.go [hunk 1]
```diff
@@ -306,9 +312,18 @@ func (b *BulkIndexer) Delete(index, _type, id string) {
...
+func (b *BulkIndexer) UpdateWithWithScript(index string, _type string, id, parent, ttl string, date *time.Time, script EsScript, scripted_upsert bool, upsert interface{}) error {
 
 	var data map[string]interface{} = make(map[string]interface{})
+	if scripted_upsert {
+		data["scripted_upsert"] = true
+	}
+	if upsert != nil {
+		data["upsert"] = upsert
+	} else {
+		data["upsert"] = map[string]interface{}{}
```
Comment (arun251) [inline lib/corebulk.go]: Does this need to be set in all cases?

Comment (zsiddique) [reply lib/corebulk.go]: No, that was suppose to be inside the scripted_upsert (if that is set to true, you must have an upsert key)

File: lib/corebulk.go [hunk 2]
```diff
@@ -306,9 +312,18 @@ func (b *BulkIndexer) Delete(index, _type, id string) {
 	return
 }
 
-func (b *BulkIndexer) UpdateWithWithScript(index string, _type string, id, parent, ttl string, date *time.Time, script string) error {
+func (b *BulkIndexer) UpdateWithWithScript(index string, _type string, id, parent, ttl string, date *time.Time, script EsScript, scripted_upsert bool, upsert interface{}) error {
```
Comment (pkaeding) [inline lib/corebulk.go]: Is it still possible to supply an inline script?  Insofar as elastigo is a general purpose golang<->elasticsearch library, we shouldn't remove any functionality from it, even if we don't need that functionality ourselves.

Comment (zsiddique) [reply lib/corebulk.go]: Ok I added the inline to the struct. The old function would actually not work with ES 2.X anyways: https://www.elastic.co/guide/en/elasticsearch/reference/2.3/modules-scripting.html

**Pull request merged**